apps/client/src/app/pages/portfolio/fire/fire-page.component.ts

```diff
@@ -149,6 +150,29 @@ export class FirePageComponent implements OnDestroy, OnInit {
       });
   }
 
+  public onRulesUpdated(event: UpdateUserSettingDto) {
+    this.isLoading = true;
+    this.dataService
+      .putUserSetting(event)
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe(() => {
+        this.dataService
```

Comment:
Could you extract this in a separate function `initializePortfolioReport()` and reuse it as well in `ngOnInit()`? You could use a dedicated variable `isLoadingPortfolioReport`...